source/fundamentals/logging.txt

```diff
@@ -0,0 +1,477 @@
+         standard operations in the MongoDB Java driver will not produce output
+         from Log4J2 without configuration. To learn how to change your Log4J2
+         logger's log level, see the 
+         :ref:`example in the Configure Your Logger section of this page <configure-log-level>`.
+
+      For more information on Log4j2, see the
+      `Log4j2 manual <https://logging.apache.org/log4j/2.x/manual/>`__.
+
+Configure Your Logger
+---------------------
+
+To configure your logger, you must use the configuration system of the logging
+framework bound to SLF4J.
+
+In the following example we show how you can use your logging framework's
+configuration system to set your logger's **log level**.
+
+A logger's log level specifies a lower bound for how urgent a message must be
+for the logger to output that message.
+
+Example - Configure
+~~~~~~~~~~~~~~~~~~~
+
+This example shows how to configure your logger's log level to INFO.
+Select the tab corresponding to the logging framework you are using in your
+project.
+
+.. _configure-log-level:
+
+.. tabs::
+
+   .. tab:: Logback
+      :tabid: Logback-binding
+
+      Specify Logback configurations in a file named ``logback.xml``. Your
+      ``logback.xml`` file does not have to be in a specific location, but it must
+      be accessible from your classpath. 
+
+      The Logback framework defines the following log levels. You can
+      see the log levels listed below in order from most urgent to least
+      urgent: 
+      
+      - ERROR
+      - WARN
+      - INFO
+      - DEBUG
+      - TRACE 
+
+      Set your ``logback.xml`` file to the following.
+
+      .. code-block:: xml
+
+         <configuration>
+         <appender name="CONSOLE"
+                   class="ch.qos.logback.core.ConsoleAppender">
+           <!-- deny all events with a level below INFO, that is TRACE and DEBUG -->
+           <filter class="ch.qos.logback.classic.filter.ThresholdFilter">
+             <level>INFO</level>
+           </filter>
+           <encoder>
+             <pattern>
+               %-4relative [%thread] %-5level %logger{30} - %msg%n
+             </pattern>
+           </encoder>
+         </appender>
+         <root level="DEBUG">
+           <appender-ref ref="CONSOLE" />
+         </root>
+         </configuration>
+
+      To test that your logger configuration was successful, run the following
+      code.
+
+      .. include:: /includes/fundamentals/logging.rst
+
+      You should see output that resembles the following.
+
+      .. code-block:: none
+
+         ...
+         1317 [cluster-ClusterId{value='<your cluster id>', description='null'}-<your connection uri>] INFO  org.mongodb.driver.cluster - Discovered replica set primary <your connection uri>
+         1568 [main] INFO  org.mongodb.driver.connection - Opened connection [connectionId{localValue:7, serverValue:<server value>}] to <your connection uri>  
```

Comment:
I'm surprised we don't see DEBUG level logging here

Reply:
Thanks for the note. Fix: https://github.com/mongodb/docs-java/pull/76/commits/352229379217d8a0fd3aaae46b2c5f5f795011f4

I believe that the Logback configuration does produce the correct output (log messages with a log level of INFO or higher) but  does does it incorrectly ([by using a filter rather than changing the root loggers log level](http://logback.qos.ch/manual/filters.html#thresholdFilter)). Great catch, and apologies for missing this.